docs/modules/integrated-clustering/pages/spring.adoc

```diff
@@ -1,716 +0,0 @@
-= Integration with Spring
```

Comment:
It seems that this page wasn't an orphan: https://docs.hazelcast.com/hazelcast/5.1/integrated-clustering/spring#hide-nav

Reply:
That page was not listed in ToC of the documentation, meaning it was a leftover while migrating the docs to the new platform, @JakeSCahill , is it possible we forget about it? It seems all the pages in integrated-clusters module are not in any nav.adoc.

Reply:
Other than `spring.adoc` we no longer need the other content. That whole module can be removed.

The content of `spring.adoc` is important and should be integrated with the content introduced in this PR to form a Spring Developer Guide.